src/createStore.js

```diff
@@ -196,9 +194,9 @@ export default function createStore(reducer, preloadedState, enhancer) {
       isDispatching = false
     }
 
-    var listeners = currentListeners = nextListeners
-    for (var i = 0; i < listeners.length; i++) {
-      listeners[i]()
+    listeners = currentListeners.slice()
```

Comment:
I've tested the use of the for loop with [this branch](https://github.com/splendido/redux/tree/avoid-useless-notifications-ter)
See changes [here](https://github.com/splendido/redux/compare/avoid-useless-notifications-bis...splendido:avoid-useless-notifications-ter?expand=1)